tests/Unit/Evolution/Systems/RadiationTransport/M1Grey/Test_M1Closure.cpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/RadiationTransport/M1Grey/M1Closure.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// Test tags used in M1Grey code
+SPECTRE_TEST_CASE("Evolution.Systems.RadiationTransport.M1Grey.M1Closure",
+                  "[Unit][M1Grey]") {
+  // Reference DataVector
+  const DataVector ref_dv(1);
+  // Create variables
+  // Input
+  Scalar<DataVector> energy_density(ref_dv);
+  tnsr::i<DataVector, 3, Frame::Inertial> momentum_density(ref_dv);
+  tnsr::I<DataVector, 3, Frame::Inertial> fluid_velocity(ref_dv);
+  Scalar<DataVector> fluid_lorentz_factor(ref_dv);
+  tnsr::ii<DataVector, 3, Frame::Inertial> spatial_metric(ref_dv);
+  // Output
+  Scalar<DataVector> closure_factor(ref_dv);
+  tnsr::II<DataVector, 3, Frame::Inertial> pressure_tensor(ref_dv);
+  Scalar<DataVector> comoving_energy_density(ref_dv);
+  Scalar<DataVector> comoving_momentum_density_normal(ref_dv);
+  tnsr::i<DataVector, 3, Frame::Inertial>
+      comoving_momentum_density_spatial(ref_dv);
+
+  // Allocate fluid/metric variables
+  for (size_t m = 0; m < 3; m++) {
+    fluid_velocity.get(m) = 0.1 * m;
+    spatial_metric.get(m, m) = 1. + 0.1 * m * m;
+    for (size_t n = m + 1; n < 3; n++) {
+      spatial_metric.get(m, n) = 0.1 * (m + n);
+    }
+  }
+  const auto det_and_inv = determinant_and_inverse(spatial_metric);
+  const auto& inv_spatial_metric = det_and_inv.second;
+  get(fluid_lorentz_factor) = 1.;
+  for (size_t m = 0; m < 3; m++) {
+    for (size_t n = 0; n < 3; n++) {
+      get(fluid_lorentz_factor) -= spatial_metric.get(m, n) *
+                                   fluid_velocity.get(m) *
+                                   fluid_velocity.get(n);
+    }
+  }
+  get(fluid_lorentz_factor) = 1. / sqrt(get(fluid_lorentz_factor));
+
+  // Initialize closure factor (as the input value is used as initial
+  // guess for the root finding algorithm).
+  get(closure_factor) = -1.;
+  // (1) Optically thick limit:
+  Scalar<DataVector> fourthird_square_fluid_lorentz_factor(ref_dv);
```

Comment:
combine this line and the next into a single statement (you don't need to separately allocate and assign)
`Scalar<DataVector> fourthird_square_fluid_lorentz_factor(4. / 3. * square(get(fluid_lorentz_factor)));`